docs/data_driven_testing.adoc

```diff
@@ -371,6 +371,19 @@ Data tables, data pipes, and variable assignments can be combined as needed:
 include::{sourcedir}/datadriven/DataSpec.groovy[tag=combined-variable-assignment]
 ----
 
+== Type Coercion for Data Variable Values
+
+Data variable values are casted to the declared parameter type using
+https://groovy-lang.org/operators.html#_coercion_operator[type coercion]. Due to that custom type conversions can be
+provided as https://groovy-lang.org/metaprogramming.html#_extension_modules[extension module] or with the help of
+the <<extensions.adoc#_use,`@Use`>> extension on the specification (as it has no effect to the `where:` block if
```

Comment:
Could you add a test to demonstrate the `@Use` extension, as a user it is really helpful if you can start from a working example.

Reply:
The example is fully working, as there is a standard coercion from `String` to `Integer` which is why I chose it as example. A plain cast would fail, with coercion it is parsed as integer.

Don't you think a full example with `@Use` would be too verbose here?
You would need the class doing the coercion and a full specification as you have to put the `@Use` annotation to the specification for it to work on the generated helper methods.

Reply:
If you worry about space in the docs we could put it in collapsible blocks, and optionally put the example in an embedded spec:
```
[%collapsible]
====
[source,groovy,indent=0]
----
include::{sourcedir}/datadriven/DataSpec.groovy[tag=type-coercion-at-use]
----
====
```
Alternatively, we could refer them to the specification code.

I recently used the ratpack documentation and found that I was quite missing example code. Think about it, if you don't want to do it, then I'll merge without the example.